core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -297,6 +313,7 @@ private synchronized void closeContext() throws IOException {
       // developers should first mark their resources as closed prior to any exceptions being
       // thrown.
       mClosed.set(true);
+      MetricsSystem.removeMetrics(TOTAL_THREAD_METRICS_NAME);
```

Comment:
Why we need to remove the metric while close context? If it should be removed, how about other metrics?

Reply:
this is because in closeContext, all the clientPool will be reset to null.
And in the metric gauge, adding null check is not safe, adding locking will impact the performance significantly.
So directly remove the metrics is a much easier choice.